Linkta-org/core
# Bugfix/auth + router + pagination

## Description
- Fixed functionality of 'Show More' and 'Show Less' buttons in the Side Nav.
- Consolidated unprotectedRoutes into publicRoutes.
- Updated path names to align with our [Naming Conventions Guide](https://github.com/Linkta-org/core/compare/bugfix/docs/NAMING_CONVENTIONS_GUIDE.md).
- Implemented a custom CustomRequest interface in firebaseAuthMiddleware to include user information & modified the isAuthorized function to use the CustomRequest type
- Attaches a user object with userId to the request

## Type of change

## How Can this be tested? Testing Plan to review this PR
UserInputList + firebaseAuthMiddleware + publicRoutes
Router:
Before login -> Navigate to /login, /signup, /forget-password, /reset-password -> see different public pages

Pagination + Auth:
1. At root directory, run npm run dev
2. After successful sign in, should see userinputs rendered on Side Nav successfully

## Checklist:
